qa/tasks/cephfs/test_nfs.py

```diff
@@ -744,3 +744,27 @@ def test_non_existent_cluster(self):
         except CommandFailedError as e:
             if e.exitstatus != errno.ENOENT:
                 raise
+
+    def test_nfs_export_with_invalid_path(self):
+        """
+        Test that nfs exports can't be created with invalid path
+        """
+        self._test_create_cluster()
+
+        # test non-existent path
+        with self.assertRaises(CommandFailedError):
+            self._create_export(
+                export_id='123', create_fs=True,
+                extra_cmd=['--pseudo-path', self.pseudo_path, '--path',
+                           '/non_existent_dir'])
+
+        # test export creation at a filepath
+        self.ctx.cluster.run(args=['sudo', 'chmod', '1777', '/mnt'])
+        self.ctx.cluster.run(args=['touch', '/mnt/testfile'])
+        with self.assertRaises(CommandFailedError):
+            self._create_export(
+                export_id='123',
+                extra_cmd=['--pseudo-path', self.pseudo_path, '--path',
+                           'testfile'])
```

Comment:
@vshankar, things look good and teuthology run is green but I'm not able to test this, for this it says 

```
Error EINVAL: Failed to create export: path testfile is invalid
```

but it should say 

```
Error EINVAL: Failed to create export: path testfile is not a dir
```

I even tried having

```
extra_cmd=['--pseudo-path', self.pseudo_path, '--path',
                           '/mnt/testfile'])
```

but again that said 

```
Error EINVAL: Failed to create export: path /mnt/testfile is invalid
```

What am i doing wrong here? is this not the way to create a file in cluster? or am i inserting wrong `--path`?

Reply:
> @vshankar, things look good and teuthology run is green but I'm not able to test this, for this it says
> 
> ```
> Error EINVAL: Failed to create export: path testfile is invalid
> ```
> 
> but it should say
> 
> ```
> Error EINVAL: Failed to create export: path testfile is not a dir

That's because you created testfile under mnt :)
```
self.ctx.cluster.run(args=['touch', '/mnt/testfile'])
```

?

Reply:
> > @vshankar, things look good and teuthology run is green but I'm not able to test this, for this it says
> > ```
> > Error EINVAL: Failed to create export: path testfile is invalid
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but it should say
> > ```
> > Error EINVAL: Failed to create export: path testfile is not a dir
> > ```
> 
> That's because you created testfile under mnt :)
> 
> ```
> self.ctx.cluster.run(args=['touch', '/mnt/testfile'])
> ```
> 
> ?

even if i use --path as `/mnt/testfile`, i get 
```
Error EINVAL: Failed to create export: path /mnt/testfile is invalid
```

Reply:
Aren't you creating the file outside ceph filesystem?

Reply:
isn't it mounted at `mnt`?

Reply:
> isn't it mounted at `mnt`?

OK. I see what you mean - test_nfs hardcodes that which is bad :/

Reply:
this has been fixed in the testcase